src/wrapped_radix_engine_toolkit/high_level/sargon_specific_types/address_union.rs

```diff
@@ -0,0 +1,449 @@
+use crate::prelude::*;
+
+#[macro_export]
+macro_rules! address_union {
+    (
+        $(
+            #[doc = $expr: expr]
+        )*
+        $union_name: ident,
+        $(
+            $variant_name: ident,
```

Comment:
`$( ... )+` is pattern for repetition, with at LEAST ONE variant (`+`... and `*` means "zero or more")